modules/aws/ssm.go

```diff
@@ -68,3 +73,127 @@ func NewSsmClientE(t testing.TestingT, region string) (*ssm.SSM, error) {
 
 	return ssm.New(sess), nil
 }
+
+// WaitForSsmInstanceE waits until the instance get registered to the SSM inventory.
+func WaitForSsmInstanceE(t testing.TestingT, awsRegion, instanceID string, timeout time.Duration) error {
+	timeBetweenRetries := 2 * time.Second
+	maxRetries := int(timeout.Seconds() / timeBetweenRetries.Seconds())
+	description := fmt.Sprintf("Waiting for %s to appear in the SSM inventory", instanceID)
+
+	key := aws.String("AWS:InstanceInformation.InstanceId")
+	filterType := aws.String("Equal")
+	values := aws.StringSlice([]string{instanceID})
+	_, err := retry.DoWithRetryE(t, description, maxRetries, timeBetweenRetries, func() (string, error) {
+		client := NewSsmClient(t, awsRegion)
+		req, resp := client.GetInventoryRequest(&ssm.GetInventoryInput{
+			Filters: []*ssm.InventoryFilter{
+				{
+					Key:    key,
+					Type:   filterType,
+					Values: values,
+				},
+			},
+		})
+
+		if err := req.Send(); err != nil {
+			return "", err
+		}
+
+		if len(resp.Entities) != 1 {
+			return "", fmt.Errorf("%s is not in the SSM inventory", instanceID)
+		}
+
+		return "", nil
+	})
+
+	return err
+}
+
+// WaitForSsmInstance waits until the instance get registered to the SSM inventory.
+func WaitForSsmInstance(t testing.TestingT, awsRegion, instanceID string, timeout time.Duration) {
+	err := WaitForSsmInstanceE(t, awsRegion, instanceID, timeout)
+	require.NoError(t, err)
+}
+
+// CheckSsmCommand checks that you can run the given command on the given instance through AWS SSM. Returns stdout and stderr.
+func CheckSsmCommand(t testing.TestingT, awsRegion, instanceID, command string, timeout time.Duration) (string, string) {
+	stdout, stderr, err := CheckSsmCommandE(t, awsRegion, instanceID, command, timeout)
+	if err != nil {
+		t.Fatalf("failed to execute '%s' on %s (%v):]\n  stdout: %#v\n  stderr: %#v", command, instanceID, err, stdout, stderr)
+	}
+	return stdout, stderr
+}
+
+type result struct {
+	Stdout string
+	Stderr string
+}
+
+// CheckSsmCommandE checks that you can run the given command on the given instance through AWS SSM. Returns the stdout, stderr and an error if one occurs.
+func CheckSsmCommandE(t testing.TestingT, awsRegion, instanceID, command string, timeout time.Duration) (string, string, error) {
+	logger.Logf(t, "Running command '%s' on EC2 instance with ID '%s'", command, instanceID)
+
+	timeBetweenRetries := 2 * time.Second
+	maxRetries := int(timeout.Seconds() / timeBetweenRetries.Seconds())
+
+	// Now that we know the instance in the SSM inventory, we can send the command
+	client, err := NewSsmClientE(t, awsRegion)
+	if err != nil {
+		return "", "", err
+	}
+	comment := "Terratest SSM"
+	documentName := "AWS-RunShellScript"
+	req, resp := client.SendCommandRequest(&ssm.SendCommandInput{
+		Comment:      &comment,
+		DocumentName: &documentName,
+		InstanceIds:  []*string{&instanceID},
+		Parameters: map[string][]*string{
+			"commands": []*string{&command},
```

Comment:
NIT: The AWS Go SDK has lots of helpers for turning these into pointers. E.g., `aws.String(comment)`. Typically easier to read if you use those for all these params.